webpack/react_app/components/TargetingHosts/TargetingHosts.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { LoadingState } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+import HostItem from './HostItem/HostItem';
+import { getTargetingHostsAction } from '../../redux/actions/TargetingHosts';
+import {
+  selectLoadingState,
+  selectErrorState,
+  selectHostsState,
+  selectRefreshState,
+} from '../../redux/selectors/TargetingHosts';
+
+const TargetingHosts = ({ data }) => {
+  const { jobInvocationId } = data;
+  const loading = useSelector(state => selectLoadingState(state));
+  const error = useSelector(state => selectErrorState(state));
+  const hosts = useSelector(state => selectHostsState(state));
+  const refresh = useSelector(state => selectRefreshState(state));
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (refresh === true) {
+      setTimeout(
+        () =>
+          dispatch(
+            getTargetingHostsAction(
+              `/job_invocations/${jobInvocationId}?format=json`
```

Comment:
This doesn't include the pagination info, maybe you can get the url with:
```js
  import getURI from 'foremanReact/common/urlHelpers`
  const url = getURI();
  url.addQuery('format', 'json');
  return `${url.pathname()}${url.search()}`;
```
 and then the component won't need the `jobInvocationId`